tests/test_Dataset.py

```diff
@@ -623,25 +623,31 @@ def __getitem__(self, item):
     assert res.features["data"].shape == (5, 3)
 
 
+NestedSize = Union[int, List["NestedSize"]]
```

Comment:
I don't understand why you need this. You will never operate on nested sizes. We always take the sum over all files in case there is some nested structure, i.e. always a single `int`.

Reply:
The `_test` function gets this as input though?

Reply:
Sure we balance based on a single int but we should make sure the nested data is passed correctly into `get_subepoch_dataset`.

Reply:
But in `_get_files_per_sub_epochs`, the `file_sizes` is not nested, it is just a single `int`. And this test here is specifically about `_get_files_per_sub_epochs`, nothing else. I think we should not make this test more complicated.

If you want to test the nested structure in some way, I would do that in a separate test. But I'm not sure if this is really needed, to have an own test for this. I don't see much potential that this can break. I usually introduce tests in case I feel that I can easily introduce a bug at some point to it, or just to easier develop it and test it directly within the test.